lib/handler/redirect.c

```diff
@@ -79,10 +85,39 @@ static void redirect_internally(h2o_redirect_handler_t *self, h2o_req_t *req, h2
     h2o_send_error_503(req, "Internal Server Error", "internal server error", 0);
 }
 
+static h2o_iovec_t replace_wildcard(h2o_mem_pool_t *pool, h2o_iovec_t target, char_vec *wildcard_chars, h2o_iovec_t replacement)
+{
+    h2o_iovec_t replaced;
+    replaced.len = target.len + wildcard_chars->size * (replacement.len - 1);
+    replaced.base = h2o_mem_alloc_pool(pool, char, replaced.len);
+    char *src = target.base;
+    char *dst = replaced.base;
+    size_t i;
+    for (i = 0; i != wildcard_chars->size; ++i) {
+        size_t plen = wildcard_chars->entries[i] - src;
+        memcpy(dst, src, plen);
+        dst += plen;
+        src += plen;
+        memcpy(dst, replacement.base, replacement.len);
+        dst += replacement.len;
+        src += 1;
+    }
+    memcpy(dst, src, target.len - (src - target.base));
+    return replaced;
+}
+
 static int on_req(h2o_handler_t *_self, h2o_req_t *req)
 {
     h2o_redirect_handler_t *self = (void *)_self;
-    h2o_iovec_t dest = h2o_build_destination(req, self->prefix.base, self->prefix.len, 1);
+
+    h2o_iovec_t prefix;
+    if (self->wildcard_chars.entries != NULL && req->authority_wildcard_match.base != NULL) {
+        prefix = replace_wildcard(&req->pool, self->prefix, &self->wildcard_chars, req->authority_wildcard_match);
```

Comment:
How about using `h2o_concat_list`? Then, we can concentrate on building the correct list, removing the burden to concatenate the string correctly.

Reply:
done in https://github.com/h2o/h2o/pull/2186/commits/508f5b3f12bb37affaba40283981a73b1565e955